src/app/router/ui/AdminRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+import { Outlet, useNavigate } from 'react-router-dom';
+
+import { Routes, RoutesPaths } from 'shared/constants/router';
+import { useAuthContext } from 'shared/contexts/authContext';
+
+export const AdminRoute = () => {
+    const { account } = useAuthContext();
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if (!account || account.role !== 'ADMIN') {
+            navigate(RoutesPaths[Routes.LOGIN], { state: { from: window.location.pathname } });
```

Comment:
I believe for case when user has an account but role is not ADMIN system should redirect user to the main page, bot to the login, wdyt?